Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+- `Major.Minor.x :` Latest version in the Major.Minor version channel release. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used to get all latest bug fixes automatically in a major.minor version. As back-compat is maintained in major.minor version, thus ensuring applications being built will not break.
+- `Major.x :` Latest version released in the Major version channel. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used by tools or library developers who need to test with every latest version for compatibility.
+
+### How it works:
+The correct version needs to be identified in case user enters pattern based version such as 2.2.x. This is how the exact needed version and its information is extracted:
+
+- [releases-index.json](https://github.com/dotnet/core/blob/master/release-notes/releases-index.json) The file containing links to all .Net Core release channels is download and read into a JSON object `releasesIndex`. This is used to find all available release channels and links to their releases.json.
+
+- The `version` input is divided into three sections:
+  - Major
+  - Minor
+  - Patch
+
+- Based on Minor versions value, the following can happen
+  - Numeric : we get the corresponding channel's (Major.Minor) releases.json link from `releasesIndex`. Then the releases.json is downloaded and parsed into an object: `channelInformation`.
+    - NOTE: in case exact version is not found, the function logs error and task will fail.
+  - x : in this case, we find the latest channel with `support-phase` != `preview` (unless `includePreviewVersions` is true). The releases.json for the selected channel is then downloaded and parsed into json object `releaseInformation`.
+    - The version with heighest `release.sdk/runtime.version` is then returned. (preview version will only be selected if `includePreviewVersions` is true)
+  - Empty: the function will log error as version needs to be given till atleast minor version. As a result the task will fail.
+
+
+- Based on Patch version value, the following can happen
+  - Empty : the function logs error and task will fail as it expects either a value or x.
+  - Exact Version (alpha numeric): a release with exact release.sdk/runtime.version will be searched in `channelInformation`.
+    - `version found:` the version information is returned
+    - `version not found:` function logs error and task will fail.
+  - x : The latest sdk/runtime version among all releases in that channel will be searched. The latest version information is then returned.
+
+## **- Support for multiple versions to be installted side by side**
+This feature was asked by users so that multiple versions can be installed using the task and they all will be available to subsequent tasks in pipeline. It is required in projects where multiple versions of SDK/runtime are required to build/run different applications.
```

Comment:
Shouldn't it be worded as "multiple projects in an application" rather than the other way round?